doc/Command-Reference.md

```diff
@@ -7961,8 +7962,292 @@ Go Back To [Beginning of the document](#) or [Beginning of this section](#waterm
+
+    Install package
+
+  Options:
+    --enable                        Set the default state of the feature to
+                                    enabled and enable feature right after
+                                    installation. NOTE: user needs to execute
+                                    "config save -y" to make this setting
+                                    persistent
+
+    --default-owner [local|kube]    Default owner configuration setting for a
+                                    feature
+
+    --from-repository TEXT          Fetch package directly from image registry
+                                    repository NOTE: This argument is mutually
+                                    exclusive with arguments: [from_tarball,
+                                    package_expr].
+
+    --from-tarball FILE             Fetch package from saved image tarball
+                                    NOTE: This argument is mutually exclusive
+                                    with arguments: [from_repository,
+                                    package_expr].
+
+    -f, --force                     Force operation by ignoring failures
+    -y, --yes                       Automatically answer yes on prompts
+    -v, --verbosity LVL             Either CRITICAL, ERROR, WARNING, INFO or
+                                    DEBUG
+
+    --skip-cli-plugin-installation  Do not install CLI plugins provided by the
+                                    package on the host OS. NOTE: In case when
+                                    package /cli/mandatory field is set to True
+                                    this option will fail the installation.
+
+    --help                          Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager install dhcp-relay==1.0.2
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager install --from-repository azure/sonic-cpu-report:latest
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager install --from-tarball sonic-docker-image.gz
+  ```
+
+**sonic-package-manager uninstall**
+
+This command uninstalls package from SONiC host. *NOTE*: this command requires elevated (root) privileges to run.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager uninstall [OPTIONS] NAME
+
+    Uninstall package
+
+  Options:
+    -f, --force          Force operation by ignoring failures
+    -y, --yes            Automatically answer yes on prompts
+    -v, --verbosity LVL  Either CRITICAL, ERROR, WARNING, INFO or DEBUG
+    --help               Show this message and exit.
+
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager uninstall dhcp-relay
+  ```
+
+**sonic-package-manager upgrade**
+
+This command upgrades package on SONiC host to a newer version. The procedure of upgrading a package will restart the corresponding service. *NOTE*: this command requires elevated (root) privileges to run.
```

Comment:
What is the diff between this and default install, that brings in latest?

Reply:
"install" command does not bring in the latest by default. "install" will look for "default-reference" for this package in "packages.json" and install from that tag.

For the difference between install and upgrade please refer to HLD - https://github.com/stepanblyschak/SONiC/blob/sonic-app-ext-3/doc/sonic-application-extention/sonic-application-extention-hld.md#sonic-package-upgrade-flow.

Reply:
Following "apt" again, i think "upgrade" should not take arguments, but instead upgrade all packages to their latest available versions (which means we maybe also need an "sonic-package-manager update" to update all repository sources.

Reply:
"sonic-package-manager update" - this requires some external source of packages.json that will be updated - this is not planned yet.

Upgrade to their latest available versions - this might not be safe, because "latest" may be incompatible. Currently user is required to explicitely specify the version he would like to have.